## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 516538`
Commit: [`cffdfc0cd323fcdc131be8055bbc7e8b01c7d050`](https://github.com/NixOS/nixpkgs/commit/cffdfc0cd323fcdc131be8055bbc7e8b01c7d050) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/cffdfc0cd323fcdc131be8055bbc7e8b01c7d050..pull/516538/head))
Merge: [`ad5e871356da9b10d3caf064f9328ab8c4d80647`](https://github.com/NixOS/nixpkgs/commit/ad5e871356da9b10d3caf064f9328ab8c4d80647)

Logs: https://github.com/MiniHarinn/nixpkgs-review-gha/actions/runs/25326406776


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>sss_code</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>sss_code</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>sss_code</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>sss_code</summary>
<pre>  HOST_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  cargo:warning=error: unable to open output file &#x27;/homeless-shelter/.cache/clang/ModuleCache/18RISFR6A0Q8A/Cocoa-M5UKYNMBKUUP.pcm&#x27;: &#x27;No such file or directory&#x27;
  cargo:warning=In file included from objc/notify.m:1:
  cargo:warning=objc/notify.h:1:9: fatal error: could not build module &#x27;Cocoa&#x27;
  cargo:warning=    1 | #import &lt;Cocoa/Cocoa.h&gt;
  cargo:warning=      |  ~~~~~~~^
  cargo:warning=error: unable to open output file &#x27;/homeless-shelter/.cache/clang/ModuleCache/18RISFR6A0Q8A/CoreServices-2KIU141CU21BS.pcm&#x27;: &#x27;No such file or directory&#x27;
  cargo:warning=error: unable to open output file &#x27;/homeless-shelter/.cache/clang/ModuleCache/18RISFR6A0Q8A/Foundation-1D2SD47NPE6H0.pcm&#x27;: &#x27;No such file or directory&#x27;
  cargo:warning=error: unable to open output file &#x27;/homeless-shelter/.cache/clang/ModuleCache/18RISFR6A0Q8A/ObjectiveC-3DB50U3ICGCMW.pcm&#x27;: &#x27;No such file or directory&#x27;
  cargo:warning=5 errors generated.

  --- stderr


  error occurred: Command env -u IPHONEOS_DEPLOYMENT_TARGET &quot;/nix/store/b27wqif15k5b2k7j13ajh401d2g4a9m6-clang-wrapper-21.1.8/bin/cc&quot; &quot;-O3&quot; &quot;-ffunction-sections&quot; &quot;-fdata-sections&quot; &quot;-fPIC&quot; &quot;-m64&quot; &quot;--target=x86_64-apple-darwin&quot; &quot;-mmacosx-version-min=14.0&quot; &quot;-Wall&quot; &quot;-Wextra&quot; &quot;-fmodules&quot; &quot;-Wno-deprecated-declarations&quot; &quot;-mmacos-version-min=14.0&quot; &quot;-o&quot; &quot;/nix/build/nix-14457-999833549/source/target/x86_64-apple-darwin/release/build/mac-notification-sys-f373649c0cc63047/out/307ac93856b42bcb-notify.o&quot; &quot;-c&quot; &quot;objc/notify.m&quot; with args cc did not execute successfully (status code exit status: 1).


warning: build failed, waiting for other jobs to finish...</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>sss_code</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>sss_code</summary>
<pre>  HOST_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  cargo:warning=error: unable to open output file &#x27;/homeless-shelter/.cache/clang/ModuleCache/2LPESB6NVVV68/Cocoa-3H6ZQ4R9G9QAO.pcm&#x27;: &#x27;No such file or directory&#x27;
  cargo:warning=In file included from objc/notify.m:1:
  cargo:warning=objc/notify.h:1:9: fatal error: could not build module &#x27;Cocoa&#x27;
  cargo:warning=    1 | #import &lt;Cocoa/Cocoa.h&gt;
  cargo:warning=      |  ~~~~~~~^
  cargo:warning=error: unable to open output file &#x27;/homeless-shelter/.cache/clang/ModuleCache/2LPESB6NVVV68/CoreServices-3BJQO8Q15IGO6.pcm&#x27;: &#x27;No such file or directory&#x27;
  cargo:warning=error: unable to open output file &#x27;/homeless-shelter/.cache/clang/ModuleCache/2LPESB6NVVV68/Foundation-3HXFEQF3GN24N.pcm&#x27;: &#x27;No such file or directory&#x27;
  cargo:warning=error: unable to open output file &#x27;/homeless-shelter/.cache/clang/ModuleCache/2LPESB6NVVV68/ObjectiveC-1QU4Z5UK6R2G2.pcm&#x27;: &#x27;No such file or directory&#x27;
  cargo:warning=5 errors generated.

  --- stderr


  error occurred: Command env -u IPHONEOS_DEPLOYMENT_TARGET &quot;/nix/store/rwsbw420hcq148hifv7fga6r4g0k1dw8-clang-wrapper-21.1.8/bin/cc&quot; &quot;-O3&quot; &quot;-ffunction-sections&quot; &quot;-fdata-sections&quot; &quot;-fPIC&quot; &quot;--target=arm64-apple-darwin&quot; &quot;-mmacosx-version-min=14.0&quot; &quot;-Wall&quot; &quot;-Wextra&quot; &quot;-fmodules&quot; &quot;-Wno-deprecated-declarations&quot; &quot;-mmacos-version-min=14.0&quot; &quot;-o&quot; &quot;/nix/build/nix-28791-3475174713/source/target/aarch64-apple-darwin/release/build/mac-notification-sys-0331c747d7119590/out/307ac93856b42bcb-notify.o&quot; &quot;-c&quot; &quot;objc/notify.m&quot; with args cc did not execute successfully (status code exit status: 1).


warning: build failed, waiting for other jobs to finish...</pre>
</details>
</details>

--- a/pkgs/by-name/ss/sss_code/package.nix
+++ b/pkgs/by-name/ss/sss_code/package.nix
@@ -4,6 +4,7 @@
   pkg-config,
   fontconfig,
   libiconv,
+  oniguruma,
   stdenv,
   libxcb,
   lib,
@@ -32,7 +33,12 @@ rustPlatform.buildRustPackage (finalAttrs: {
   buildInputs = [
     fontconfig
     libxcb
+    oniguruma
   ];
+
+  env = {
+    RUSTONIG_SYSTEM_LIBONIG = true;
+  };
 
   doCheck = false;
 

--- a/pkgs/by-name/ss/sss_code/package.nix
+++ b/pkgs/by-name/ss/sss_code/package.nix
@@ -4,6 +4,7 @@
   pkg-config,
   fontconfig,
   libiconv,
+  writableTmpDirAsHomeHook,
   oniguruma,
   stdenv,
   libxcb,
@@ -28,7 +29,15 @@ rustPlatform.buildRustPackage (finalAttrs: {
     "sss_code"
   ];
 
-  nativeBuildInputs = [ pkg-config ] ++ lib.optionals stdenv.buildPlatform.isDarwin [ libiconv ];
+  nativeBuildInputs = [
+    pkg-config
+  ]
+  ++ lib.optionals stdenv.buildPlatform.isDarwin [
+    libiconv
+  ]
+  ++ lib.optionals stdenv.cc.isClang [
+    writableTmpDirAsHomeHook
+  ];
 
   buildInputs = [
     fontconfig

